### PR TITLE
Internalize Shader Graph API

### PR DESCRIPTION
**Pull request by pbbastian**
### Purpose of this PR
This PR internalizes all of Shader Graph's API. We have a new API coming for custom nodes that allows for more freedom than CodeFunctionNode, but doesn't rely on the internals of Shader Graph.

This PR also fixes various instances of the `UnityEditor.ShaderGraph` namespace being used in other packages.

In order to allow HDRP and LWRP to continue using Shader Graph even though the API is now internal, their Editor assemblies have been added to a special list inside Shader Graph that gives them access to internal API.

---
### Release Notes
This makes more sense to add to once the new custom node API is in.

---
### Testing status
**Katana Tests**: All green

https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=sg%2Finternalize-api&automation-tools_branch=add-platform-filter&unity_branch=trunk

**Manual Tests**: Open each test project and verify that it loads without warning, as well as open up some Shader Graphs.

**Automated Tests**: Nothing new.

---
### Overall Product Risks
**Technical Risk**: Low

Touches a lot of code, but really only renames. I've done my best to add `FormerNameAttribute` to all serialized classes that changed namespace.

**Halo Effect**: Low

Any code using Shader Graph API will have to fork Shader Graph and add themselves to the list in `com.unity.shadergraph/Editor/AssemblyInfo.cs`, as there won't be any public API left for now.

### CODE REVIEW

**Comment by spaztronaut** [comment]
@pbbastian Currently getting all nodes incompatible error in the graph and rendering pink

**Comment by pbbastian** [comment]
@spaztronaut I have fixed the issue now :)

File: com.unity.render-pipelines.high-definition/Editor/ShaderGraph/HDSubShaderUtilities.cs [hunk 1]
```diff
@@ -476,7 +476,7 @@ public struct Pass
         public string ColorMaskOverride;
         public List<string> StencilOverride;
         public List<string> RequiredFields;         // feeds into the dependency analysis
-        public ShaderGraphRequirements requirements;
+//        public ShaderGraphRequirements requirements;
```
Comment (Kink3d) [inline com.unity.render-pipelines.high-definition/Editor/ShaderGraph/HDSubShaderUtilities.cs]: Remove this?

File: com.unity.shadergraph/Editor/Data/Nodes/FunctionMultiInput.cs [hunk 2]
```diff
@@ -1,6 +1,6 @@
 namespace UnityEditor.ShaderGraph
 {
-    /*public abstract class FunctionMultiInput : BaseMaterialNode, IGeneratesBodyCode
+    /*abstract class FunctionMultiInput : BaseMaterialNode, IGeneratesBodyCode
```
Comment (Kink3d) [inline com.unity.shadergraph/Editor/Data/Nodes/FunctionMultiInput.cs]: Note: Remove this file, ignore for this PR though.

File: com.unity.render-pipelines.high-definition/Editor/ShaderGraph/HDSubShaderUtilities.cs (hunk 1 above)
Comment (pbbastian) [reply com.unity.render-pipelines.high-definition/Editor/ShaderGraph/HDSubShaderUtilities.cs]: ah yeah, definitely should. It wasn't used anywhere, and was causing issues after internalizing

Comment (pbbastian) [reply com.unity.render-pipelines.high-definition/Editor/ShaderGraph/HDSubShaderUtilities.cs]: ✔️

**Pull request merged**